src/servers/grpc_server_v2.cc

```diff
@@ -1315,24 +1362,163 @@ InferGRPCToInput(
                 .c_str());
       }
 
-      // FIXMEV2 handle non-raw content types
+      // Try to read the raw contents if available
       const std::string& raw = io.contents().raw_contents();
       base = raw.c_str();
       byte_size = raw.size();
+
+      // Check the presence of explicit tensors
+      if (io.contents().bool_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "BOOL", io.datatype(), byte_size));
+        base = (const void*)io.contents().bool_contents().data();
+        byte_size = io.contents().bool_contents_size();
+      }
+
+      if (io.contents().int_contents_size() != 0) {
+        if (io.datatype().compare("INT8") == 0) {
+          RETURN_IF_ERR(InferGRPCToInputHelper(
+              io.name(), request.model_name(), "INT8", io.datatype(),
+              byte_size));
+          std::shared_ptr<std::string> serialized(new std::string());
+          serialized_data_map->emplace(io.name(), serialized);
+          for (const auto& element : io.contents().int_contents()) {
+            // Assuming the system is little-endian, picking the
+            // least significant byte of 32-bit integer as a
+            // int8 element
+            serialized->append(reinterpret_cast<const char*>(&element), 1);
```

Comment:
Instead of magic numbers you should get the size of the datatype once above and use it as needed. We should have a utility to get the size from the V2 datatypes... you might need to add that.

Reply:
Or for performance reasons you can just keep it like it is but put comments explaining each magic number. Like "1 /* sizeof INT8 */"

Reply:
I have added the utility in src/servers/common.cc